cmd/geth/snapshot.go

```diff
@@ -576,3 +626,356 @@ func dumpState(ctx *cli.Context) error {
+				values := make([][]byte, 256)
+				chunkkey := trieUtils.GetTreeKeyCodeChunk(accIt.Hash().Bytes(), uint256.NewInt(uint64(i)))
+				j := i
+				for ; (j-i) < 256 && j < len(chunks); j++ {
+					values[(j-128)%256] = chunks[j][:]
+				}
+				i = j
+
+				// Otherwise, store the previous group in the tree with a
+				// stem insertion.
+				treeHuggers[int(chunkkey[0])/rootPerCPU] <- &treeHugger{stem: chunkkey[:31], node: verkle.NewLeafNode(chunkkey[:31], values)}
+			}
+
+			// Write the code size in the account header group
+			var size [32]byte
+			newValues[4] = size[:]
+			binary.LittleEndian.PutUint64(size[:8], uint64(len(code)))
+		}
+
+		// Save every slot into the tree
+		if !bytes.Equal(acc.Root, emptyRoot[:]) {
+			var (
+				laststem [31]byte
+				values   = make([][]byte, 256)
+			)
+			copy(laststem[:], stem)
+
+			storageIt, err := snaptree.StorageIterator(root, accIt.Hash(), common.Hash{})
+			if err != nil {
+				log.Error("Failed to open storage trie", "root", acc.Root, "error", err)
+				return err
+			}
+			for storageIt.Next() {
+				slotkey := trieUtils.GetTreeKeyStorageSlot(accIt.Hash().Bytes(), uint256.NewInt(0).SetBytes(storageIt.Slot()))
+
+				var value [32]byte
+				copy(value[:len(storageIt.Slot())-1], storageIt.Slot())
+
+				// if the slot belongs to the header group, store it there
+				if bytes.Equal(slotkey[:31], stem) {
+					newValues[int(slotkey[31])] = value[:]
+					continue
+				}
+
+				// if the slot belongs to the same group as the previous
+				// one, add it to the current group of values.
+				if bytes.Equal(laststem[:], slotkey[:31]) {
+					values[slotkey[31]] = value[:]
+					continue
+				}
+
+				// flush the previous group, iff it's not the header group
+				if !bytes.Equal(stem[:31], laststem[:]) {
+					treeHuggers[int(laststem[0])/rootPerCPU] <- &treeHugger{stem: laststem[:], node: verkle.NewLeafNode(laststem[:], values)}
+				}
+			}
+			if !bytes.Equal(laststem[:31], stem[:31]) {
+				treeHuggers[int(laststem[0])/rootPerCPU] <- &treeHugger{stem: laststem[:], node: verkle.NewLeafNode(laststem[:], values)}
+			}
+			storageIt.Release()
+			if storageIt.Error() != nil {
+				log.Error("Failed to traverse storage trie", "root", acc.Root, "error", storageIt.Error())
+				return storageIt.Error()
+			}
+			// Finish with storing the complete account header group
+			// inside the tree.
+			treeHuggers[int(stem[0])/rootPerCPU] <- &treeHugger{stem: stem[:], node: verkle.NewLeafNode(stem[:], newValues)}
+		}
+
+		if time.Since(lastReport) > time.Second*8 {
+			log.Info("Traversing state", "accounts", accounts, "elapsed", common.PrettyDuration(time.Since(start)))
+			lastReport = time.Now()
+		}
+	}
+	if accIt.Error() != nil {
+		log.Error("Failed to compute commitment", "root", root, "error", accIt.Error())
+		return accIt.Error()
+	}
+	log.Info("Wrote all leaves", "accounts", accounts, "elapsed", common.PrettyDuration(time.Since(start)))
+	for _, hugger := range treeHuggers {
+		close(hugger)
```

Comment:
Closing is a good start, but you also should wait for them to exit. Otherwise you have 256 channels, with 128 things, in total ~32K  buffered items which you just ignore.

Reply:
good catch, I'll change that. Note, though, that there are only `runtime.NumCPU()` channels.